[wasm] Globalization fails when hosting an app using `data:` uri

### Current behavior

Updating `Uno.UI.WebAssembly` from version `2.9.0-dev.224` to `2.9.0-dev.240` ([this PR](https://github.com/unoplatform/uno/pull/6906)) causes the following problem when using an uno app from `data:` storage:

```
uno-bootstrap.js:210 Error: System.TypeInitializationException: The type initializer for 'Windows.Globalization.ApplicationLanguages' threw an exception.
 ---> System.Exception: Failed to read the 'localStorage' property from 'Window': Storage is disabled inside 'data:' URLs.
Error: Failed to read the 'localStorage' property from 'Window': Storage is disabled inside 'data:' URLs.
    at ApplicationDataContainer.tryGetValue (http://localhost:21064/package_2826c6c9bb4b79431228694055ef1f5faf0e99a8/Uno.UI.js:3066:17)
    at dispatch (http://localhost:21064/package_2826c6c9bb4b79431228694055ef1f5faf0e99a8/Uno.UI.js:294:55)
    at _mono_wasm_invoke_js_unmarshalled (http://localhost:21064/package_2826c6c9bb4b79431228694055ef1f5faf0e99a8/dotnet.js:1:202209)
    at wasm://wasm/00978b42:wasm-function[219]:0x1a101
    at wasm://wasm/00978b42:wasm-function[167]:0xcad3
    at wasm://wasm/00978b42:wasm-function[166]:0xb9e6
    at wasm://wasm/00978b42:wasm-function[2810]:0xab7c2
    at wasm://wasm/00978b42:wasm-function[1615]:0x6f707
    at wasm://wasm/00978b42:wasm-function[1619]:0x6fd71
    at wasm://wasm/00978b42:wasm-function[1616]:0x6f9da
   at Uno.Foundation.WebAssemblyRuntime.InnerInvokeJSUnmarshalled(String functionIdentifier, IntPtr arg0, IntPtr arg1) in C:\a\1\s\src\Uno.Foundation.Runtime.WebAssembly\Interop\Runtime.wasm.cs:line 197
   at Uno.Foundation.WebAssemblyRuntime.InvokeJSUnmarshalled(String functionIdentifier, IntPtr arg0, IntPtr arg1) in C:\a\1\s\src\Uno.Foundation.Runtime.WebAssembly\Interop\Runtime.wasm.cs:line 163
   at Uno.Foundation.Interop.TSInteropMarshaller.InvokeJS(String methodName, Object paramStruct, Type retStructType, String memberName) in C:\a\1\s\src\Uno.Foundation.Runtime.WebAssembly\Interop\TSInteropMarshaller.wasm.cs:line 94
   at Windows.Storage.ApplicationDataContainerInterop.TryGetValue(ApplicationDataLocality locality, String key, String& value) in /home/vsts/work/1/s/src/Uno.UWP/Storage/ApplicationDataContainer.wasm.cs:line 231
   at Windows.Storage.ApplicationDataContainer.FilePropertySet.TryGetValue(String key, Object& value) in /home/vsts/work/1/s/src/Uno.UWP/Storage/ApplicationDataContainer.wasm.cs:line 153
   at Windows.Globalization.ApplicationLanguages..cctor() in /home/vsts/work/1/s/src/Uno.UWP/Globalization/ApplicationLanguages.cs:line 23
   --- End of inner exception stack trace ---
   at Windows.UI.Xaml.Application.Start(ApplicationInitializationCallback callback) in /home/vsts/work/1/s/src/Uno.UI/UI/Xaml/Application.cs:line 254
   at Uno.Figma.Plugin.Editor.Program.Main(String[] args) in C:\src\Uno.Figma.PlugIn\src\Uno.Figma.Plugin.Editor\Program.cs:line 19
    at Object.call_method (http://localhost:21064/package_2826c6c9bb4b79431228694055ef1f5faf0e99a8/uno-bootstrap.js:162:19)
    at Object.mainInit (http://localhost:21064/package_2826c6c9bb4b79431228694055ef1f5faf0e99a8/uno-bootstrap.js:204:29)
```

This is crashing the application while booting. Doesn't seem to have a workaround.

## Interpretation of the problem
It seems the problem is not caused directly by the PR, but this one is causing a type initialization at a moment where it cannot be handled correctly. The fix should be to adjust the javascript `UnoStatic_Windows_Storage_ApplicationDataContainer:tryGetValue()` to handle the case where there is no access to file system.

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

--- a/src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts
+++ b/src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts
@@ -30,6 +30,7 @@ namespace Windows.Storage {
 			} catch (e) {
 				ret.Value = "";
 				ret.HasValue = false;
+				console.debug(`ApplicationDataContainer.tryGetValue failed: ${e}`);
 			}
 
 			ret.marshal(pReturn);
@@ -48,6 +49,7 @@ namespace Windows.Storage {
 
 				localStorage.setItem(storageKey, params.Value);
 			} catch (e) {
+				console.debug(`ApplicationDataContainer.setValue failed: ${e}`);
 			}
 
 			return true;
@@ -66,6 +68,7 @@ namespace Windows.Storage {
 				ret.ContainsKey = localStorage.hasOwnProperty(storageKey);
 			} catch (e) {
 				ret.ContainsKey = false;
+				console.debug(`ApplicationDataContainer.containsKey failed: ${e}`);
 			}
 
 			ret.marshal(pReturn);
@@ -98,7 +101,7 @@ namespace Windows.Storage {
 					}
 				}
 			} catch (e) {
-
+				console.debug(`ApplicationDataContainer.getKeyByIndex failed: ${e}`);
 			}
 
 			ret.Value = returnKey;
@@ -127,7 +130,7 @@ namespace Windows.Storage {
 					}
 				}
 			} catch (e) {
-
+				console.debug(`ApplicationDataContainer.getCount failed: ${e}`);
 			}
 
 			ret.marshal(pReturn);
@@ -158,7 +161,7 @@ namespace Windows.Storage {
 					localStorage.removeItem(itemsToRemove[item]);
 				}
 			} catch (e) {
-
+				console.debug(`ApplicationDataContainer.clear failed: ${e}`);
 			}
 
 			return true;
@@ -177,6 +180,7 @@ namespace Windows.Storage {
 				ret.Removed = localStorage.hasOwnProperty(storageKey);
 			} catch (e) {
 				ret.Removed = false;
+				console.debug(`ApplicationDataContainer.remove failed: ${e}`);
 			}
 
 			if (ret.Removed) {
@@ -213,7 +217,7 @@ namespace Windows.Storage {
 					}
 				}
 			} catch (e) {
-
+				console.debug(`ApplicationDataContainer.getValueByIndex failed: ${e}`);
 			}
 
 			ret.Value = returnKey;

--- a/src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts
+++ b/src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts
@@ -10,34 +10,6 @@ namespace Windows.Storage {
 			return `UnoApplicationDataContainer_${locality}_`;
 		}
 
-		// https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#feature-detecting_localstorage
-		private static isLocalStorageAvailable() {
-			let storage: Storage;
-			try {
-				storage = window['localStorage'];
-				const x = "__storage_test__";
-				storage.setItem(x, x);
-				storage.removeItem(x);
-				return true;
-			} catch (e) {
-				return (
-					e instanceof DOMException &&
-					// everything except Firefox
-					(e.code === 22 ||
-						// Firefox
-						e.code === 1014 ||
-						// test name field too, because code might not be present
-						// everything except Firefox
-						e.name === "QuotaExceededError" ||
-						// Firefox
-						e.name === "NS_ERROR_DOM_QUOTA_REACHED") &&
-					// acknowledge QuotaExceededError only if there's something already stored
-					storage &&
-					storage.length !== 0
-				);
-			}
-		}
-
 		/**
 		 * Try to get a value from localStorage
 		 * */
@@ -47,10 +19,15 @@ namespace Windows.Storage {
 
 			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			if (ApplicationDataContainer.isLocalStorageAvailable() && localStorage.hasOwnProperty(storageKey)) {
-				ret.HasValue = true;
-				ret.Value = localStorage.getItem(storageKey);
-			} else {
+			try {
+				if (localStorage.hasOwnProperty(storageKey)) {
+					ret.HasValue = true;
+					ret.Value = localStorage.getItem(storageKey);
+				} else {
+					ret.Value = "";
+					ret.HasValue = false;
+				}
+			} catch (e) {
 				ret.Value = "";
 				ret.HasValue = false;
 			}
@@ -64,12 +41,13 @@ namespace Windows.Storage {
 		 * Set a value to localStorage
 		 * */
 		private static setValue(pParams: number): boolean {
-			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+			try {
 				const params = ApplicationDataContainer_SetValueParams.unmarshal(pParams);
 
 				const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
 				localStorage.setItem(storageKey, params.Value);
+			} catch (e) {
 			}
 
 			return true;
@@ -84,7 +62,11 @@ namespace Windows.Storage {
 
 			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			ret.ContainsKey = ApplicationDataContainer.isLocalStorageAvailable() && localStorage.hasOwnProperty(storageKey);
+			try {
+				ret.ContainsKey = localStorage.hasOwnProperty(storageKey);
+			} catch (e) {
+				ret.ContainsKey = false;
+			}
 
 			ret.marshal(pReturn);
 
@@ -102,7 +84,7 @@ namespace Windows.Storage {
 			let returnKey = "";
 			const prefix = ApplicationDataContainer.buildStoragePrefix(params.Locality);
 
-			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+			try {
 				for (let i = 0; i < localStorage.length; i++) {
 					const storageKey = localStorage.key(i);
 
@@ -115,6 +97,8 @@ namespace Windows.Storage {
 						localityIndex++;
 					}
 				}
+			} catch (e) {
+
 			}
 
 			ret.Value = returnKey;
@@ -134,7 +118,7 @@ namespace Windows.Storage {
 			ret.Count = 0;
 			const prefix = ApplicationDataContainer.buildStoragePrefix(params.Locality);
 
-			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+			try {
 				for (let i = 0; i < localStorage.length; i++) {
 					const storageKey = localStorage.key(i);
 
@@ -142,6 +126,8 @@ namespace Windows.Storage {
 						ret.Count++;
 					}
 				}
+			} catch (e) {
+
 			}
 
 			ret.marshal(pReturn);
@@ -159,7 +145,7 @@ namespace Windows.Storage {
 
 			const itemsToRemove: string[] = [];
 
-			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+			try {
 				for (let i = 0; i < localStorage.length; i++) {
 					const storageKey = localStorage.key(i);
 
@@ -171,6 +157,8 @@ namespace Windows.Storage {
 				for (const item in itemsToRemove) {
 					localStorage.removeItem(itemsToRemove[item]);
 				}
+			} catch (e) {
+
 			}
 
 			return true;
@@ -185,7 +173,11 @@ namespace Windows.Storage {
 
 			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			ret.Removed = ApplicationDataContainer.isLocalStorageAvailable() && localStorage.hasOwnProperty(storageKey);
+			try {
+				ret.Removed = localStorage.hasOwnProperty(storageKey);
+			} catch (e) {
+				ret.Removed = false;
+			}
 
 			if (ret.Removed) {
 				localStorage.removeItem(storageKey);
@@ -207,7 +199,7 @@ namespace Windows.Storage {
 			let returnKey = "";
 			const prefix = ApplicationDataContainer.buildStoragePrefix(params.Locality);
 
-			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+			try {
 				for (let i = 0; i < localStorage.length; i++) {
 					const storageKey = localStorage.key(i);
 
@@ -220,6 +212,8 @@ namespace Windows.Storage {
 						localityIndex++;
 					}
 				}
+			} catch (e) {
+
 			}
 
 			ret.Value = returnKey;

--- a/src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts
+++ b/src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts
@@ -10,6 +10,34 @@ namespace Windows.Storage {
 			return `UnoApplicationDataContainer_${locality}_`;
 		}
 
+		// https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#feature-detecting_localstorage
+		private static isLocalStorageAvailable() {
+			let storage: Storage;
+			try {
+				storage = window['localStorage'];
+				const x = "__storage_test__";
+				storage.setItem(x, x);
+				storage.removeItem(x);
+				return true;
+			} catch (e) {
+				return (
+					e instanceof DOMException &&
+					// everything except Firefox
+					(e.code === 22 ||
+						// Firefox
+						e.code === 1014 ||
+						// test name field too, because code might not be present
+						// everything except Firefox
+						e.name === "QuotaExceededError" ||
+						// Firefox
+						e.name === "NS_ERROR_DOM_QUOTA_REACHED") &&
+					// acknowledge QuotaExceededError only if there's something already stored
+					storage &&
+					storage.length !== 0
+				);
+			}
+		}
+
 		/**
 		 * Try to get a value from localStorage
 		 * */
@@ -19,7 +47,7 @@ namespace Windows.Storage {
 
 			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			if (localStorage.hasOwnProperty(storageKey)) {
+			if (ApplicationDataContainer.isLocalStorageAvailable() && localStorage.hasOwnProperty(storageKey)) {
 				ret.HasValue = true;
 				ret.Value = localStorage.getItem(storageKey);
 			} else {
@@ -36,11 +64,13 @@ namespace Windows.Storage {
 		 * Set a value to localStorage
 		 * */
 		private static setValue(pParams: number): boolean {
-			const params = ApplicationDataContainer_SetValueParams.unmarshal(pParams);
+			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+				const params = ApplicationDataContainer_SetValueParams.unmarshal(pParams);
 
-			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
+				const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			localStorage.setItem(storageKey, params.Value);
+				localStorage.setItem(storageKey, params.Value);
+			}
 
 			return true;
 		}
@@ -54,7 +84,7 @@ namespace Windows.Storage {
 
 			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			ret.ContainsKey = localStorage.hasOwnProperty(storageKey);
+			ret.ContainsKey = ApplicationDataContainer.isLocalStorageAvailable() && localStorage.hasOwnProperty(storageKey);
 
 			ret.marshal(pReturn);
 
@@ -72,16 +102,18 @@ namespace Windows.Storage {
 			let returnKey = "";
 			const prefix = ApplicationDataContainer.buildStoragePrefix(params.Locality);
 
-			for (let i = 0; i < localStorage.length; i++) {
-				const storageKey = localStorage.key(i);
+			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+				for (let i = 0; i < localStorage.length; i++) {
+					const storageKey = localStorage.key(i);
 
-				if (storageKey.startsWith(prefix)) {
+					if (storageKey.startsWith(prefix)) {
 
-					if (localityIndex === params.Index) {
-						returnKey = storageKey.substr(prefix.length);
+						if (localityIndex === params.Index) {
+							returnKey = storageKey.substr(prefix.length);
+						}
+
+						localityIndex++;
 					}
-
-					localityIndex++;
 				}
 			}
 
@@ -102,11 +134,13 @@ namespace Windows.Storage {
 			ret.Count = 0;
 			const prefix = ApplicationDataContainer.buildStoragePrefix(params.Locality);
 
-			for (let i = 0; i < localStorage.length; i++) {
-				const storageKey = localStorage.key(i);
+			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+				for (let i = 0; i < localStorage.length; i++) {
+					const storageKey = localStorage.key(i);
 
-				if (storageKey.startsWith(prefix)) {
-					ret.Count++;
+					if (storageKey.startsWith(prefix)) {
+						ret.Count++;
+					}
 				}
 			}
 
@@ -125,16 +159,18 @@ namespace Windows.Storage {
 
 			const itemsToRemove: string[] = [];
 
-			for (let i = 0; i < localStorage.length; i++) {
-				const storageKey = localStorage.key(i);
+			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+				for (let i = 0; i < localStorage.length; i++) {
+					const storageKey = localStorage.key(i);
 
-				if (storageKey.startsWith(prefix)) {
-					itemsToRemove.push(storageKey);
+					if (storageKey.startsWith(prefix)) {
+						itemsToRemove.push(storageKey);
+					}
 				}
-			}
 
-			for (const item in itemsToRemove) {
-				localStorage.removeItem(itemsToRemove[item]);
+				for (const item in itemsToRemove) {
+					localStorage.removeItem(itemsToRemove[item]);
+				}
 			}
 
 			return true;
@@ -149,7 +185,7 @@ namespace Windows.Storage {
 
 			const storageKey = ApplicationDataContainer.buildStorageKey(params.Locality, params.Key);
 
-			ret.Removed = localStorage.hasOwnProperty(storageKey);
+			ret.Removed = ApplicationDataContainer.isLocalStorageAvailable() && localStorage.hasOwnProperty(storageKey);
 
 			if (ret.Removed) {
 				localStorage.removeItem(storageKey);
@@ -171,16 +207,18 @@ namespace Windows.Storage {
 			let returnKey = "";
 			const prefix = ApplicationDataContainer.buildStoragePrefix(params.Locality);
 
-			for (let i = 0; i < localStorage.length; i++) {
-				const storageKey = localStorage.key(i);
+			if (ApplicationDataContainer.isLocalStorageAvailable()) {
+				for (let i = 0; i < localStorage.length; i++) {
+					const storageKey = localStorage.key(i);
 
-				if (storageKey.startsWith(prefix)) {
+					if (storageKey.startsWith(prefix)) {
 
-					if (localityIndex === params.Index) {
-						returnKey = localStorage.getItem(storageKey);
+						if (localityIndex === params.Index) {
+							returnKey = localStorage.getItem(storageKey);
+						}
+
+						localityIndex++;
 					}
-
-					localityIndex++;
 				}
 			}
 

